sessionctx/varsutil/varsutil.go

```diff
@@ -25,6 +25,23 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+// formatSQLModeStr re-formate 'SQL_MODE' variable.
+func formatSQLModeStr(s string) (ret string) {
+	parts := strings.Split(s, ",")
+	parts[len(parts)-1] = strings.TrimSpace(parts[len(parts)-1])
+	for i := 0; i < len(parts); i++ {
+		if len(parts[i]) == 0 {
+			continue
+		}
+		if len(ret) == 0 {
+			ret = parts[i]
+		} else {
+			ret = ret + "," + parts[i]
+		}
+	}
```

Comment:
How about collect non-empty parts to another slice, then call `strings.Join`?
I think that would be easier to read.

Reply:
ok